pkg/patch/patch_test.go

```diff
@@ -78,6 +81,49 @@ func TestGeneratePatch_AppendSidecarTrue(t *testing.T) {
 	verifyPatch(t, string(patch), meta)
 }
 
+func TestGeneratePatch_AppendSidecarTrue_WithFargateProfile(t *testing.T) {
+	meta := Meta{
+		AppendImagePullSecret: false,
+		HasImagePullSecret:    false,
+		AppendSidecar:         true,
+		AppendInit:            false,
+		Init: InitMeta{
+			Ports:              "80,443",
+			EgressIgnoredPorts: "22",
+			ContainerImage:     "111345817488.dkr.ecr.us-west-2.amazonaws.com/aws-appmesh-proxy-route-manager:v2",
+			IgnoredIPs:         "169.254.169.254",
+		},
+		Sidecar: SidecarMeta{
+			MeshName:          "global",
+			VirtualNodeName:   "podinfo",
+			Region:            "us-west-2",
+			LogLevel:          "debug",
+			ContainerImage:    "111345817488.dkr.ecr.us-west-2.amazonaws.com/aws-appmesh-envoy:latest",
+			CpuRequests:       "10m",
+			MemoryRequests:    "32Mi",
+			InjectXraySidecar: true,
+			EnableStatsTags:   true,
+		},
+		PodMetadata: metav1.ObjectMeta{
+			Labels: map[string]string{
+				config.FargateProfileLabel: "some-profile",
```

Comment:
Might be useful to have a test case that has both  "appmesh.k8s.aws/appmeshCNI: disabled" and "eks.amazonaws.com/fargate-profile: some-profile" and verifies that the appmesh.k8s.aws/appmeshCNI annotation is not added even though it's a Fargate pod.